tests/handlers/github_test.py

```diff
@@ -0,0 +1,151 @@
+"""Test the github webhook handler."""
+
+import hashlib
+import hmac
+from dataclasses import dataclass
+from pathlib import Path
+from string import Template
+
+import pytest
+import respx
+from httpx import AsyncClient
+
+from ..support.constants import TEST_GITHUB_WEBHOOK_SECRET
+from ..support.gafaelfawr import mock_gafaelfawr
+
+
+@dataclass(frozen=True)
+class GithubRequest:
+    payload: str
+    headers: dict[str, str]
+
+
+def webhook_request(org: str, repo: str, ref: str) -> GithubRequest:
+    """Build a Github webhook request and headers with the right hash."""
+    data_path = Path(__file__).parent.parent / "data"
+    template = (data_path / "github_webhook.tmpl.json").read_text()
+    payload = Template(template).substitute(
+        org=org,
+        repo=repo,
+        ref=ref,
+    )
+
+    # https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries#python-example
+    hash_object = hmac.new(
+        TEST_GITHUB_WEBHOOK_SECRET.encode("utf-8"),
+        msg=payload.encode("utf-8"),
```

Comment:
Minor nit: Since UTF-8 is the default, I generally just write `encode()` and save a few characters and a tiny bit of mental horsepower trying to figure out if that charset is doing anything special.